src/pkg/generators/host_generator.go

```diff
@@ -188,13 +189,12 @@ func (h *hostsSimulator) Generate(
 					labels = append(labels, h.appendLabels...)
 				}
 
-				timestamp := now
+				timestampUnixMillis := now.UnixMilli()
 				if h.coldWritesPercent > 0 && rand.Float64() <= h.coldWritesPercent {
-					timestamp = now.Add(-time.Duration(float64(h.coldWritesRange) * rand.Float64()))
+					timestamp := now.Add(-time.Duration(float64(h.coldWritesRange) * rand.Float64()))
+					timestampUnixMillis = timestamp.Truncate(scrapeDuration).UnixMilli()
 				}
 
-				timestampUnixMillis := timestamp.Truncate(30*time.Second).UnixNano() / int64(time.Millisecond)
```

Comment:
Now it no longer aligns warm write timestamps to scrape interval, is that intended?

Reply:
Yes. This was the original behaviour before I introduced cold write support in https://github.com/m3dbx/promremotebench/pull/22. I realized that this truncation only makes sense when generating cold writes (to prevent an unbounded growth of datapoint density). For warm writes, it is not needed.